### ISSUE TITLE
Attribute names in /v2/items

### ISSUE THREAD

**Issue by dr-ishmael** [arenanet/api-cdi#112]
The new attributes Concentration and Expertise are being returned on the items endpoint `attributes` and `buff` as `BoonDuration` and `ConditionDuration`, the same as actual Boon Duration and Condition Duration. Luckily, on equipment (weapons/armor/trinkets/back) they can be distinguished because Concentration/Expertise actually show up in `attributes`, whereas Boon/Condition Duration only show up in `buff`.
- Boon Duration on equipment: https://api.guildwars2.com/v2/items/38168
- Concentration on equipment: https://api.guildwars2.com/v2/items/72391

However, upgrade components only have a `buff`, so we have to look to see if the number has a percent sign or not before we can translate `Boon/ConditionDuration` to Concentration/Expertise.
- Boon Duration on upgrade: https://api.guildwars2.com/v2/items/38143
- Concentration on upgrade: https://api.guildwars2.com/v2/items/73386

Is there any way these attribute names can be updated?

PS: the attribute Ferocity is still being returned as `CritDamage`, although this isn't a major problem because it doesn't create any conflicts.

**Comment by lye** [comment]
I mean, they're enumeration values that have been there for a long time, so we can't change them without breaking existing applications. The `Boon/ConditionDuration` enum value has been in there for awhile; I'm not sure if it's entirely safe to change.

The "Boon Duration +X%" buff is the old-style before it was set up as an actual attribute -- it was implemented as a hidden buff that modified boon/condi applications when you used a skill. When we added durations as actual attributes, the old stuff didn't get updated. There are two completely separate ways to modify boon/condi durations, and the API reflects this. I agree that it's kinda gross, but I don't really want to mess with it until I sneak `/v2/itemz` past @tivac.

**Comment by dr-ishmael** [comment]
I understand that about the buffs - back when Magic Find was an equipment
attribute, it was built the same way. But you're saying that the percentage
Boon Duration and the non-percentage Concentration share the same
enumeration?

**Comment by tivac** [comment]
>  but I don't really want to mess with it until I sneak /v2/itemz past @tivac.

ಠ\_ಠ

**Comment by StevenLiekens** [comment]
> but I don't really want to mess with it until I sneak /v2/itemz past @tivac.
> ಠ_ಠ

ಠ◡ಠ

**Comment by lye** [comment]
> But you're saying that the percentage Boon Duration and the non-percentage Concentration share the same enumeration?

Nah, the percentage Boon Duration is not a character attribute. It's not even an enumeration value -- I'm pretty sure it's just a string (it's even [localized](https://api.guildwars2.com/v2/items/38168?lang=de)!). :sadness:

**Comment by dr-ishmael** [comment]
Interesting. I'm at home now and checked some of these in-game - the upgrade components, like [Freshwater Pearl](https://api.guildwars2.com/v2/items/76179) and [Exquisite Freshwater Pearl](https://api.guildwars2.com/v2/items/73386), *actually show* `+9 Boon Duration`, just like it is in the `buff` on the API. But the trinkets like [Freshwater Pearl Orichalcum Earring](https://api.guildwars2.com/v2/items/72391), get transformed to `+35 Concentration`.

And NOW I read your edit and see that you already know about it. Ah well, I took the time to write all that, I'm posting it anyway. :)

**Comment by lye** [comment]
I think what I want to do is add an additional field to the `details` object that contains the machine-readable jewel stats. I don't want to mess with the existing fields for fear of application breakage. I'll have to write a PR later.

**Comment by Archomeda** [comment]
Recently the Boon Duration and Condition Duration stats on (as far as I know, all) equipment have been changed to their non-percentage equivalent, so I think this is no longer an issue?

**Comment by Aonwy** [comment]
If there is additional need around this, please write up a new ticket.
